hydrus/data/resource_based_classes.py

```diff
@@ -0,0 +1,209 @@
+"""
+Script to generate the tables in hydrus database based on
+resources in the provided API Doc.
+"""
+import copy
+from hydrus.data.db_models import Resource
+from hydrus.data.exceptions import (
+    ClassNotFound,
+    DatabaseConstraintError,
+    InstanceExists,
+    InstanceNotFound,
+    InvalidSearchParameter,
+    PropertyNotFound,
+    PropertyNotGiven,
+)
+from sqlalchemy import exists
+from sqlalchemy.orm.scoping import scoped_session
+from sqlalchemy.orm.exc import NoResultFound
+
+from typing import Dict, Any
+
+
+def get_type(object_: Dict[str, Any]) -> str:
+    """
+    Return the @type of that given object.
+    :param object_: Dict containing object properties
+    :return: The @type of that object
+    """
+    return object_["@type"]
+
+
+def get_database_class(type_: str):
+    """
+    Get the sqlalchemy class object from given classname
+    :param type_: The @type of a object
+    :return: The SQL-Alchemy database class object for that type_
+    """
+    database_class = Resource.all_database_classes.get(type_, None)
+    if database_class is None:
+        raise ClassNotFound(type_)
+    return database_class
+
+
+def insert_object(object_: Dict[str, Any], session: scoped_session) -> str:
+    """
+    Insert the object in the database
+    :param object_: Dict containing object properties
```

Comment:
From what I can see the `object_` dictionary is the same as the one passed in the request right? In that case the property resource class name and the key in `object_` may not be same.

For example in `Drone` we can have `DroneState` object mapped to the "state" key in the request, but the table name would be DroneState. Where are you mapping the keys to the actual resource classes?

Reply:
Yes, the `object_` dictionary is the same as the one passed in the request.
From what I understood, your doubt is that in the request we would pass the id of a **State** instance in the key called `DroneState`. But the table in the database is called `State`, so where am I mapping the `DroneState` key to the `State` table instance.

That I am doing in [here](https://github.com/HTTP-APIs/hydrus/blob/27db73939f2fe8a8d08a8e3c2096876247dcf7d0/hydrus/data/resource_based_classes.py#L63). While creating the tables itself, I am adding a dict in the `.info` attribute of any foreign key through which I can extract the column name, which then helps in extracting the correct value from the `object_` dict.

Does that make it clear?